utilities/rpp-unittests/SOW3_OCL/BatchPD_ocl_pln1.cpp

```diff
@@ -0,0 +1,850 @@
+
+        Rpp8u *ip_image = image.data;
+        Rpp8u *ip_image_second = image_second.data;
+        for (j = 0; j < srcSize[i].height; j++)
+        {
+            for (int x = 0; x < srcSize[i].width; x++)
+            {
+                for (int y = 0; y < ip_channel; y++)
+                {
+                    input[count + ((j * maxWidth * ip_channel) + (x * ip_channel) + y)] = ip_image[(j * srcSize[i].width * ip_channel) + (x * ip_channel) + y];
+                    input_second[count + ((j * maxWidth * ip_channel) + (x * ip_channel) + y)] = ip_image_second[(j * srcSize[i].width * ip_channel) + (x * ip_channel) + y];
+                }
+            }
+        }
+        i++;
+    }
+    closedir(dr2);
+
+    if (ip_bitDepth == 1)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp16f *inputf16Temp, *inputf16_secondTemp;
+
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf16Temp = inputf16;
+        inputf16_secondTemp = inputf16_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf16Temp = ((Rpp16f)*inputTemp) / 255.0;
+            *inputf16_secondTemp = ((Rpp16f)*input_secondTemp) / 255.0;
+            inputTemp++;
+            inputf16Temp++;
+            input_secondTemp++;
+            inputf16_secondTemp++;
+        }
+    }
+    else if (ip_bitDepth == 2)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp32f *inputf32Temp, *inputf32_secondTemp;
+
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf32Temp = inputf32;
+        inputf32_secondTemp = inputf32_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf32Temp = ((Rpp32f)*inputTemp) / 255.0;
+            *inputf32_secondTemp = ((Rpp32f)*input_secondTemp) / 255.0;
+            inputTemp++;
+            inputf32Temp++;
+            input_secondTemp++;
+            inputf32_secondTemp++;
+        }
+    }
+    else if (ip_bitDepth == 5)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp8s *inputi8Temp, *inputi8_secondTemp;
+
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputi8Temp = inputi8;
+        inputi8_secondTemp = inputi8_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputi8Temp = (Rpp8s)(((Rpp32s)*inputTemp) - 128);
+            *inputi8_secondTemp = (Rpp8s)(((Rpp32s)*input_secondTemp) - 128);
+            inputTemp++;
+            inputi8Temp++;
+            input_secondTemp++;
+            inputi8_secondTemp++;
+        }
+    }
+
+    cl_mem d_input, d_inputf16, d_inputf32, d_inputi8, d_input_second, d_output, d_outputf16, d_outputf32, d_outputi8;
+    cl_platform_id platform_id;
+    cl_device_id device_id;
+    cl_context theContext;
+    cl_command_queue theQueue;
+    cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)